pytket_qir/gatesets/pyqir/pyqir.py

```diff
@@ -42,6 +42,7 @@
 PYQIR_GATES = CustomGateSet(
     name="PyQir",
     template=Template("__quantum__${opnat}__${opname}__${opspec}"),
+    base_gateset=_TK_TO_PYQIR.keys() | set(),
```

Comment:
```suggestion
    base_gateset=set(_TK_TO_PYQIR.keys()),
```
Simpler, and makes it always a `set`.

Reply:
Thanks, but why always ? The pipe operator won't work if the `keys()` array is empty ?

Reply:
> Thanks, but why always ? The pipe operator won't work if the `keys()` array is empty ?

I don't think we need the pipe operator? `keys()` does not return a `set` so with your version we get different types depending if it is empty or not.

Reply:
Oh really ? Hm OK thanks for the clarification then.